hr_holidays_hour/models/hr_holidays.py

```diff
@@ -0,0 +1,256 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
+# Copyright 2019 Coop IT Easy SCRLfs
+#   - Vincent Van Rossem <vincent@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import datetime
+import logging
+
+from openerp import _, api, fields, models
+from openerp.osv import osv
+from openerp.exceptions import ValidationError
+from openerp.exceptions import Warning as UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class HrHolidays(osv.osv):
+    _inherit = "hr.holidays"
+
+    # workaround to override constraints
+
+    def _check_date(self, cr, uid, ids, context=None):
+        return True
+
+    _check_holidays = lambda self, cr, uid, ids, context=None: self.check_holidays(
+        cr, uid, ids, context=context
+    )
+
+    _constraints = [
+        (
+            _check_date,
+            "You can not have 2 leaves that overlaps on same day!",
+            ["date_from", "date_to"],
+        ),
+        (
+            _check_holidays,
+            "The number of remaining leaves is not sufficient for this leave type.\n"
+            "Please verify also the leaves waiting for validation.",
+            ["state", "number_of_days_temp"],
+        ),
+    ]
+
+    def check_holidays(self, cr, uid, ids, context=None):
+        return True
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    number_of_hours_temp = fields.Float(
+        string="Allocation in Hours",
+        digits=(2, 2),
+        readonly=True,
+        states={
+            "draft": [("readonly", False)],
+            "confirm": [("readonly", False)],
+        },
+    )
+    number_of_hours = fields.Float(
+        compute="_compute_number_of_hours", store=True
+    )
+    virtual_hours = fields.Float(
+        compute="_compute_number_of_hours", store=True
+    )
+    working_hours = fields.Float(digits=(2, 2))
+
+    @api.onchange("employee_id")
+    def onchange_employee(self):
+        # Get result of original onchange_employee from parent class:
+        res = super(HrHolidays, self).onchange_employee(self.employee_id.id)
+
+        # Workaround for api incompatibility:
+        if type(res) is dict and res.has_key("value"):
+            for field, value in res.get("value").items():
+                if hasattr(self, field):
+                    setattr(self, field, value)
+
+        # Additional code
+        self.department_id = None
+        self.number_of_hours_temp = 0.0
+        if self.employee_id:
+            self._set_number_of_hours_temp()
+            self.department_id = self.employee_id.department_id
+
+    @api.onchange("date_from", "date_to")
+    def onchange_date(self):
+        # Check in context what form is open: add or remove
+        if self.env.context.get("default_type", "") == "add":
+            return
+
+        self._check_dates()
+        self._check_employee()
+        self._set_number_of_hours_temp()
```

Comment:
I don't think it's ok to use a multi method from an onchange method. The database cursor will be rolled back (_I think_)

Reply:
indeed onchange don't persist data. not sure it still apply on 12 though.